components/arm/fake/fake.go

```diff
@@ -125,6 +131,22 @@ type Arm struct {
 	model      referenceframe.Model
 }
 
+// UpdateAction helps hinting the reconfiguration process on what strategy to use given a modified config.
+// See config.UpdateActionType for more information.
+func (a *Arm) UpdateAction(c *config.Component) config.UpdateActionType {
+	if _, ok := c.ConvertedAttributes.(*AttrConfig); ok {
+		if model, err := buildModel(*c); err != nil {
+			a.logger.Debugf(
+				`%v - we continue using current model: %v`,
+				err.Error(), a.model.Name())
+		} else {
+			a.joints = &pb.JointPositions{Values: make([]float64, len(a.model.DoF()))}
+			a.model = model
+		}
+	}
+	return config.None
```

Comment:
why not rebuild?

Reply:
agreed - I had not thought through the implications of using `config.Rebuild` instead for this case.

Let's make it explicit.
given
```
_, ok := c.ConvertedAttributes.(*AttrConfig)
```
Supposing `!ok` we return `config.Rebuild`.
We will eventually have to rebuild the fake arm through the `NewArm` method. This would then call `buildModel(cfg)` which then does:
```
armModel := cfg.ConvertedAttributes.(*AttrConfig).ArmModel
modelPath := cfg.ConvertedAttributes.(*AttrConfig).ModelFilePath
```
but since `!ok`, we would error in the `buildModel(cfg)` method. 
This is the correct way to notify the user of an improper attribute cfg.

Adding test case for this now.